app-modules/task/src/Policies/TaskPolicy.php

```diff
@@ -99,10 +99,11 @@ public function update(Authenticatable $authenticatable, Task $task): Response
             return Response::deny('You do not have permission to update this task.');
         }
 
-        return $authenticatable->canOrElse(
-            abilities: ["task.{$task->getKey()}.update"],
-            denyResponse: 'You do not have permission to update this task.'
-        );
```

Comment:
I think we still need what you deleted here after the `assigned_to` and `created_by` check instead of `Response::allow()` because the User should still have to have `test.*.update` which this will cover.